### ISSUE TITLE
ConnectAsyncV4IPEndPointToDualHost_Succes failed in CI on Linux

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#3682]
http://dotnet-ci.cloudapp.net/job/dotnet_corefx_linux_debug_tst_prtest/3091/console
```
System.Net.Sockets.Tests.DualMode.ConnectAsyncV4IPEndPointToDualHost_Success [FAIL]
16:16:16       System.Net.Sockets.SocketException : Unknown error 10054
16:16:16       Stack Trace:
16:16:16             at System.Net.Sockets.Tests.DualMode.DualModeConnectAsync_IPEndPointToHost_Helper(IPAddress connectTo, IPAddress listenOn, Boolean dualModeServer)
16:16:16             at System.Net.Sockets.Tests.DualMode.ConnectAsyncV4IPEndPointToDualHost_Success()
```
cc: @cipop, @pgavlin, @davidsh

**Comment by stephentoub** [comment]
Failed again here: http://dotnet-ci.cloudapp.net/job/dotnet_corefx_linux_debug_tst_prtest/3154/console
```
System.Net.Sockets.Tests.DualMode.ConnectAsyncV4IPEndPointToDualHost_Success [FAIL]
19:20:29       System.Net.Sockets.SocketException : Unknown error 10054
19:20:29       Stack Trace:
19:20:29             at System.Net.Sockets.Tests.DualMode.DualModeConnectAsync_IPEndPointToHost_Helper(IPAddress connectTo, IPAddress listenOn, Boolean dualModeServer)
19:20:29             at System.Net.Sockets.Tests.DualMode.ConnectAsyncV4IPEndPointToDualHost_Success()
19:20:29             at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor)
19:20:29             at System.Reflection.RuntimeMethodInfo.UnsafeInvokeInternal(Object obj, Object[] parameters, Object[] arguments)
19:20:29             at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
```

**Comment by ericeil** [comment]
Looks like the same underlying cause as #3494.

**Comment by ericeil** [comment]
Closed with #5907
